examples/typescript/platform-ref-aws/crossplane.ts

```diff
@@ -0,0 +1,125 @@
+  });
+
+  config.addProvider('registry.upbound.io/crossplane/provider-aws', '>=v0.14.0-0');
+  config.addProvider('registry.upbound.io/crossplane/provider-helm', '>=v0.3.6-0');
+
+  return yaml;
+}
+
+export function postgresYaml(crossplanePackage: App): Chart {
+  const yaml = new Chart(crossplanePackage, 'postgres-api');
+  const xrd = new CompositeResourceDefinition(yaml, 'postgres-xrd', {
+    name: 'compositepostgresqlinstances.aws.platformref.crossplane.io',
+  });
+
+  xrd.group('aws.platformref.crossplane.io');
+  xrd.claimKind('PostgreSQLInstance').plural('postgresqlinstances');
+  xrd.kind('CompositePostgreSQLInstance').plural('compositepostgresqlinstances');
+  xrd.connectionSecret().key('username').key('password').key('endpoint').key('port');
+  xrd.version('v1alpha1').served().referencable()
+    .spec().with(Prop.for({ object: (spec) => {
+      spec.propObject('parameters').required().with(Prop.for({ object: (params) => {
+        params.uiSection( { 
+          title: 'Database Size', 
+          description: 'Enter information to size your database'});
+
+        params.propInteger('storageGB').required().min(1).max(500)
+          .description('GB of storage for your database')
+          .uiInput({
+            name: 'storageGB',
+            inputType: InputType.SINGLE_INPUT,
+            title: 'Storage (GB)',
+            default: 5
+          })
+
+        params.propObject('networkRef').required()
+          .description('A reference to the Network object that this postgres should be connected to.')
+          .with(Prop.for({ object: (networkRef) => {
```

Comment:
This will not translate well in jsii

Reply:
Yeah, jsii doesn't play nice with callback funcs as args so this was the recommended workaround. Not great, but given it's optional would like to see how this translates to other languages.